File: frontend/src/lib/types/network.ts
export type StationNode = {
	id: string;
	code: string;
	name: string;
	type: string;
	lat: number;
	lon: number;
	platforms: number;
	tracks: number;
	dailyTrains: number;
};

export type DirectionalState = {
	status: string; // 'active' | 'blocked' | 'restricted'
	restrictedVmax: number | null;
	activeEventId: string | null;
};

export type TrackSegment = {
	segmentId: string;
	source: string;
	target: string;
	line: number;
	distKm: number;
	travelMin: number;
	vmax: number;
	railTracks: number;
	forward: DirectionalState; // stan relacji source -> target
	backward: DirectionalState; // stan relacji target -> source
};

export type NetworkGraph = {
	stations: StationNode[];
	segments: TrackSegment[];
	relationshipCount: number;
};

File: backend/app/schemas/network.py
from pydantic import BaseModel


class StationNode(BaseModel):
	id: str
	code: str
	name: str
	type: str
	lat: float
	lon: float
	platforms: int
	tracks: int
	dailyTrains: int


class DirectionalState(BaseModel):
	status: str  # 'active' | 'blocked' | 'restricted'
	restrictedVmax: int | None = None
	activeEventId: str | None = None


class TrackSegment(BaseModel):
	segmentId: str
	source: str
	target: str
	line: int
	distKm: float
	travelMin: int
	vmax: int
	railTracks: int
	forward: DirectionalState  # stan relacji source -> target
	backward: DirectionalState  # stan relacji target -> source


class NetworkGraphResponse(BaseModel):
	stations: list[StationNode]
	segments: list[TrackSegment]
	relationshipCount: int
